src/transport/PeerMessageCounter.h

```diff
@@ -113,9 +113,9 @@ class PeerMessageCounter
 
         // 2. Counter Window check
         uint32_t offset = mSynced.mMaxCounter - counter;
-        if (offset < CHIP_CONFIG_MESSAGE_COUNTER_WINDOW_SIZE)
+        if (offset <= CHIP_CONFIG_MESSAGE_COUNTER_WINDOW_SIZE)
```

Comment:
Here, have you changed the meaning of the `mWindow` by shifting its represented bit by 1 ? Would you please update the comment of the member fields in `Synced` class.

You may also have changed the meaning of `SetCounter` function. Before this changes, `SetCounter(X)` will accept a packet with counter X, but with the change, it won't accept a packet with counter X. I'm not sure if it is expected.

Reply:
> Here, I think you have changed the meaning of the mWindow by shifting its represented bit by 1 ?

Yes, that is what the spec requires... Note that this is not being changed for unicast yet.

> Before this changes, SetCounter(X) will accept a packet with counter X

Not relevant for group counters, because those never use `SetCounter`.  But when we try to fix the off-by-one problem for unicast counters, we will definitely need to figure this part out...

Reply:
Well while this is true that it change the current behaviour, it simply put it up to spec for group counter. The same change needs to be applied to Unicast counter since the spec specify at section 4.5.4.1 that the bit 0 of the receiving window represent `MaxCounter -1` and not `MaxCounter`. See issue : #15455 @bzbarsky-apple